ibm/data_source_ibm_tg_gateway.go

```diff
@@ -0,0 +1,68 @@
+package ibm
+
+import (
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func dataSourceIBMTransitGateway() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceIBMTransitGatewayRead,
+
+		Schema: map[string]*schema.Schema{
+
+			tgID: {
```

Comment:
Generally we write datasources to take the name of the resource ..
Try to fetch all gateways and match with name

Reply:
ok.. changed to take name instead of id.